pkg/scheduler/framework/interface.go

```diff
@@ -160,6 +162,29 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// StatusEqual returns whether two Statuses are equal to each other.
+func StatusEqual(x, y *Status) bool {
+	if x == y {
+		return true
+	}
+	if x == nil || y == nil {
+		return x.IsSuccess() && y.IsSuccess()
+	}
+	if x.code != y.code {
+		return false
+	}
+	if x.code == Error {
+		return cmp.Equal(x.err, y.err, cmpopts.EquateErrors())
+	}
+	return len(x.reasons) == 0 && len(y.reasons) == 0
+}
+
+// Equal checks equality of two statuses. This is useful for testing with
+// cmp.Equal.
+func (s *Status) Equal(x *Status) bool {
```

Comment:
Does this imply we can only use `cmp.Equal()` (instead of `cmd.Diff()`) in UTs to get a boolean result (instead of verbosed diff)?

Reply:
That's the price to pay when giving a comparer (as opposed to exposing the fields).

Maybe a [transformer](https://pkg.go.dev/github.com/google/go-cmp/cmp#Transformer) can yield to better diffs than a comparer. For example, the transformer could do something like:

```golang
if status.code == Success {return nil}
if status.code == Error {return status.err}
else {return map[string]interface{}{"code": status.code, "reasons": status.reasons}
```

Reply:
`cmp.Equal()` looks good to me, given that UT failures are usually easy to debug - I'm just confirming my understanding.

Reply:
> cmp.Equal() looks good to me, given that UT failures are usually easy to debug - I'm just confirming my understanding.

+1, if we only need Status comparer for our tests,   `(s *Status) Equal(x *Status) bool` should be enough considering we only care about code and error. Then we can drop the comparer option.